code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -123,3 +123,104 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+	//How far can we dash
+	var/range = 5
+	//Can we use the ability again
+	var/recast_available = FALSE
+	//Is this the recast
+	var/recast = FALSE
+	//The last tile we dashed through, used when swapping with a human
+	var/turf/last_turf
+
+/datum/action/xeno_action/activable/acid_dash/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+/datum/action/xeno_action/activable/acid_dash/on_cooldown_finish()
+	to_chat(owner, span_xenodanger("Our exoskeleton quivers as we get ready to use Acid Dash again."))
+	playsound(owner, "sound/effects/xeno_newlarva.ogg", 50, 0, 1)
+	return ..()
+
+/datum/action/xeno_action/activable/acid_dash/proc/dash_complete()
+	SIGNAL_HANDLER
+	if(recast_available && !recast)
+		addtimer(CALLBACK(src, /datum/action/xeno_action/proc/add_cooldown), 2 SECONDS) //Recast 'window', how long you have to press the ability button again
+		recast = TRUE
+	else
+		recast_available = FALSE
+		recast = FALSE
+		add_cooldown()
+	UnregisterSignal(owner, list(COMSIG_XENO_OBJ_THROW_HIT, COMSIG_XENO_NONE_THROW_HIT, COMSIG_XENO_LIVING_THROW_HIT, COMSIG_MOVABLE_MOVED))
+
+/datum/action/xeno_action/activable/acid_dash/proc/mob_hit(datum/source, mob/M)
+	SIGNAL_HANDLER
+	if(recast) //That's the recast, we don't stop for mobs
+		return COMPONENT_KEEP_THROWING
+	if(ishuman(M)) //It's the first cast, we swap with the first human in our path
+		var/mob/living/carbon/human/target = M
+		var/mob/living/carbon/xenomorph/X = owner
+		var/turf/victim_turf = get_turf(target)
+		var/turf/aggressor_turf = last_turf
+
+		target.throw_at(aggressor_turf, 1, 1, X)
+		X.throw_at(victim_turf, 1, 1, X) //They swap tile
+		target.ParalyzeNoChain(0.5 SECONDS) //Extremely brief, we don't want them to take 289732 ticks of acid
+
+		to_chat(target, span_highdanger("The [X] tackles us, sending us behind them!"))
+		X.visible_message(span_xenodanger("\The [X] tackles [target], swapping location with them!"), \
+			span_xenodanger("We push [target] in our acid trail!"), visible_message_flags = COMBAT_MESSAGE)
+
+		recast_available = TRUE
+		return
+	return COMPONENT_KEEP_THROWING
+
+/datum/action/xeno_action/activable/acid_dash/proc/obj_hit(datum/source, obj/target, speed)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/X = owner
+	if(istype(target, /obj/structure/table) || istype(target, /obj/structure/barricade) || istype(target, /obj/structure/razorwire)) //Some flavour so it doesn't look completely ridiculous
+		var/obj/structure/S = target
+		X.visible_message(span_danger("[X] effortlessly jumps over the [S]!"), null, null, 5)
+	else
+		target.hitby(X, speed) //This resets throwing.
+
+//Drops an acid puddle on the current owner's tile, will do 0 damage if the owner has no acid_spray_damage
+/datum/action/xeno_action/activable/acid_dash/proc/acid_steps(atom/A, atom/OldLoc, Dir, Forced)
+	SIGNAL_HANDLER
+	last_turf = OldLoc
+	var/mob/living/carbon/xenomorph/X = owner
+	var/turf/current_turf = get_turf(X)
```

Comment:
var doesn't look useful

Reply:
If you mean last_turf, I can't do the swapping stuff without it because mob_hit is called the moment you're on the same tile as the mob, so I can't swap without lots of pointless logic.
Otherwise I can replace current_turf with get_turf sure.

Reply:
i mean current turf